internal/storage/storage.go

```diff
@@ -143,18 +165,30 @@ func newS3Backend(config S3Conf) *s3Backend {
 		}
 	}
 
-	return &s3Backend{
+	sb := &s3Backend{
 		Bucket: config.Bucket,
 		Uploader: s3manager.NewUploader(s3Session, func(u *s3manager.Uploader) {
 			u.PartSize = int64(config.Chunksize)
 			u.Concurrency = config.UploadConcurrency
 			u.LeavePartsOnError = false
 		}),
 		Client: s3.New(s3Session)}
+
+	_, err = sb.Client.ListObjectsV2(&s3.ListObjectsV2Input{Bucket: &config.Bucket})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return sb, nil
 }
 
 // NewFileReader returns an io.Reader instance
 func (sb *s3Backend) NewFileReader(filePath string) (io.ReadCloser, error) {
+	if sb == nil {
+		return nil, fmt.Errorf("Invalid s3Backend")
+	}
+
```

Comment:
Is this necessary?An invalid back-end should have been caught at startup and prevented this point from being reached.

Reply:
Should (of course) not be needed, no. With that said, these are cheap in the normal case and simplifies test writing a bit (or rather understanding if things fail).

My understanding of what happened is that a later crash made the prior assert not show so it took longer than it should have to figure out what happened.

I have a slight preference for leaving them because of that but am totally fine with pulling them out if you don't like them.

Reply:
As far as I know, it is a good practice to have this. It is not a crucial component, but still makes sure that you can track down the error better.